### PR TITLE
AAF Adapter: support for reading markers

### LINKED ISSUE #307: AAF adapter should read markers/locators
The AAF adapter currently ignores markers/locators. It should turn them into Marker objects on the appropriate Clip, Track, or Stack.

### PR DESCRIPTION
**Pull request by timlehr**
**Link the Issue(s) this Pull Request is related to.**

Fixes #307

**Summarize your change.**

This PR is based on the work done in #832.

Markers now support color properly and are attached to the appropriate item (e.g. clip, gap etc.) on the track. 

AAF metadata `set` and `dict` values are now also correctly translated instead of being converted to a string representation.

**Reference associated tests.**

- added new tests to `test_aaf_adapter.py`

### CODE REVIEW

**Comment by jminor** [comment]
This looks great! I'll try running some of our production AAFs through to see if the result looks as expected.

The CI failures seem to be just related to the linter. You can run the linter locally with `make lint` and I think it will pick the same linter used in the CI.

**Comment by fabal** [comment]
Hey @jminor @ssteinbach , sorry I had to pull the branch out as we need to wait a bit more for the final legal stuff. We'll get back to you soon.

**Comment by timlehr** [comment]
@jminor the branch is back up and also includes the linting fixes!

File: contrib/opentimelineio_contrib/adapters/advanced_authoring_format.py
Comment (jminor) [inline contrib/opentimelineio_contrib/adapters/advanced_authoring_format.py:1435]: TODO: Let's check to see if this is left over from a previous merge/change. It seems unrelated to the marker feature.

Comment (jminor) [reply contrib/opentimelineio_contrib/adapters/advanced_authoring_format.py:1435]: Yes, this looks like a reversion of some code due to this PR changing hands several times. In video-conference code review with @timlehr he says he can fix this.

**Review by jminor — CHANGES_REQUESTED** [review]

File: contrib/opentimelineio_contrib/adapters/advanced_authoring_format.py
Comment (timlehr) [reply contrib/opentimelineio_contrib/adapters/advanced_authoring_format.py:1435]: Hey Josh, I was off on vacation, but I will look into this soon.

Comment (jminor) [reply contrib/opentimelineio_contrib/adapters/advanced_authoring_format.py:1435]: Thanks @timlehr - let us know if we can help.

**Comment by timlehr** [comment]
@jminor I rebased the PR and updated the tests with a new marker file.

File: contrib/opentimelineio_contrib/adapters/tests/test_aaf_adapter.py
Comment (jminor) [inline contrib/opentimelineio_contrib/adapters/tests/test_aaf_adapter.py:985]: Is this print statement intentional, or left over from debugging?

**Review by jminor — COMMENTED** [review]
This looks great @timlehr ! I left one question inline. I'm testing it out now.

File: contrib/opentimelineio_contrib/adapters/tests/test_aaf_adapter.py
Comment (timlehr) [reply contrib/opentimelineio_contrib/adapters/tests/test_aaf_adapter.py:985]: Good catch. That's definitely just a debug print!

**Comment by timlehr** [comment]
Can you approve @fabal for the list of people who can run CI tests? I think Github requires approval for each person that doesn't have a merged commit yet. It's a bit annoying, since Fabrice has to push my commits from AL, but the CI is not triggered when he does that.

**Comment by jminor** [comment]
It looks like the tests are complaining about this:
```
 The schema has changed and the autogenerated documentation in /home/runner/work/OpenTimelineIO/OpenTimelineIO/docs/tutorials/otio-plugins.md needs to be updated.  run: `make doc-plugins-update`
```

Also, in manual testing, I'm finding some markers are not in the right offset relative to their clip. Let me get more details about that so we can find the issue.

**Comment by jminor** [comment]
I pushed the button to approve action/workflow runs for this PR. I think it will automatically run for further commits.

**Comment by jminor** [comment]
In the test AAF included in this PR, I see that all of the markers are correct except one. Here are screenshots from Avid Media Composer and otioview showing the one that is offset. It looks like it is in a nested clip. It is possible that the offset is related to that nesting. I'll find some other examples to see if this holds true.
 
Screen Shot 2021-11-16 at 12 45 56 PM

Screen Shot 2021-11-16 at 12 51 12 PM

Also here's a text summary:
```
  MARKER: global: 00:00:00:00 local: 00:00:00:00 duration: 1.0 colo: CYAN name: INSERT
  MARKER: global: 00:00:00:00 local: 00:00:00:00 duration: 1.0 colo: RED name: PUBLISH
  MARKER: global: 00:00:00:00 local: 00:00:00:00 duration: 1.0 colo: YELLOW name: FX
  MARKER: global: 00:00:04:10 local: 00:00:04:10 duration: 1.0 colo: YELLOW name: YELLOW: V5: Digital Header: In gap
  MARKER: global: 00:00:12:07 local: 00:00:45:23 duration: 1.0 colo: GREEN name: GREEN: V1: zts02_1010: f1104: seq.f1104
  MARKER: global: 00:00:21:14 local: 00:00:21:14 duration: 1.0 colo: BLUE name: BLUE: V2 (no FX): zts02_1020: f1134: seq.f1327
  MARKER: global: 00:00:35:16 local: 00:00:35:16 duration: 1.0 colo: CYAN name: CYAN: V3: zts02_1030: f1212: seq.f1665
  MARKER: global: 00:00:35:17 local: 01:00:00:00 duration: 1.0 colo: MAGENTA name: MAGENTA: V4: zts02_1040: f1001: seq.f1666
  MARKER: global: 00:00:52:20 local: 00:00:44:20 duration: 1.0 colo: RED name: RED: V5: zts02_1050: f1061: seq.f1885
  MARKER: global: 00:01:14:19 local: 00:01:14:19 duration: 1.0 colo: GREEN name: GREEN: TC1: zts02_1080: f1206: seq.f2604
```

generated via this code:
```python
for child in timeline.each_child():
    top_level = child
    while top_level.parent() is not None:
        top_level = top_level.parent()
    for marker in child.markers:
        print("  MARKER: global: {} local: {} duration: {} colo: {} name: {}".format(
            otio.opentime.to_timecode(child.transformed_time(marker.marked_range.start_time, top_level)),
            otio.opentime.to_timecode(marker.marked_range.start_time),
            marker.marked_range.duration.value,
            marker.color,
            marker.name
        ))
```

File: contrib/opentimelineio_contrib/adapters/advanced_authoring_format.py
Comment (jminor) [inline contrib/opentimelineio_contrib/adapters/advanced_authoring_format.py:1207]: I think `transformed_time` could handle this in one step:
`marked_start_local = current_track.transformed_time(marker.marked_range.start_time, target_item)`

**Review by jminor — APPROVED** [review]

**Comment by jminor** [comment]
Thanks for the fix. I verified that the marker offsets are correct for all of my sample AAFs.
Once we get the tests to pass, we can land this.

File: docs/tutorials/otio-plugins.md
Comment (jminor) [inline docs/tutorials/otio-plugins.md:338]:
```suggestion
Reads AAF content from `filepath` and outputs an OTIO
```

**Comment by jminor** [comment]
Thanks @timlehr and @fabal !

**Pull request merged**
